cmd/thanos/receive.go

```diff
@@ -94,6 +95,9 @@ func registerReceive(app *extkingpin.App) {
 			"This can trigger compaction without those blocks and as a result will create an overlap situation. Set it to true if you have vertical compaction enabled and wish to upload blocks as soon as possible without caring"+
 			"about order.").
 		Default("false").Hidden().Bool()
+	hashFunc := cmd.Flag("shipper.hash-func",
+		"Specify which hash function to use when calculating the hashes of produced files. If no function has been specified, it does not happen. This permits avoiding downloading some files twice albeit at some performance cost. Possible values are: \"\", \"SHA256\".").
+		Default("").Enum("", "SHA256")
```

Comment:
As this flag is used in multiple places, maybe we should extract out it in a function in `cmd/thanos/flags.go`.

Reply:
I agree, let's do it. However, how should we call it? Simply `--hash-func`? WDYT @prmsrswt, @bwplotka ?

Reply:
I guess `--hash-func` would work.